x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/EsqlUnsupportedOperationException.java

```diff
@@ -8,42 +8,40 @@
 package org.elasticsearch.xpack.esql;
 
 import org.elasticsearch.xpack.ql.QlServerException;
-import org.elasticsearch.xpack.ql.type.DataType;
 
+/**
+ * Indicates an unsupported operation like methods missing an implementation or code that should be unreachable or dead.
+ * Throws an error during construction if assertions are enabled. This is important for integration tests because we want to crash the
+ * server with an error in case we encounter unsupported operations, but do not want to crash in production.
+ */
 public class EsqlUnsupportedOperationException extends QlServerException {
     public EsqlUnsupportedOperationException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
         super(message, cause, enableSuppression, writableStackTrace);
+        assert false : message;
     }
 
     public EsqlUnsupportedOperationException(String message, Throwable cause) {
         super(message, cause);
+        assert false : message;
     }
 
     public EsqlUnsupportedOperationException(String message, Object... args) {
         super(message, args);
+        assert false : message;
     }
 
     public EsqlUnsupportedOperationException(Throwable cause, String message, Object... args) {
         super(cause, message, args);
+        assert false : message;
```

Comment:
I'd suggest this so we get a properly-formatted message and whole stack trace for the cause:

```suggestion
        assert false : new AssertionError(getMessage(), cause);
```

(likewise in the other places with either a formattable message or a cause)